src/agent/coverage/src/block/linux.rs

```diff
@@ -130,60 +150,46 @@ impl Recorder {
     }
 
     fn on_module_load(&mut self, tracee: &mut Tracee, image: &ModuleImage) -> Result<()> {
-        log::info!("module load: {}", image.path().display());
-
-        let blocks = find_module_blocks(image.path())?;
+        log::info!("module load: {}", image.path());
 
-        log::debug!("found {} blocks", blocks.len());
+        // Fetch disassembled module info via cache.
+        let info = self.modules.fetch(image.path())?;
```

Comment:
Rather than handling this option via is_none+bail, followed by unwrap later on, use `ok_or_else` to convert the missing fetch to a failure.

```suggestion
        let info = self.modules.fetch(image.path())?.ok_or_else(|| format_err("unable to fetch info for module: {}", image.path()))?;
```